README.md

```diff
@@ -265,6 +265,14 @@ It's useful in several situations, e.g.:
 - Joining separate changes that make more sense when applied together
 - Rewriting _work in progress_ commits
 
+### When to avoid rebase or squash?
+
+Both rebase and squash commands substitutes existing commits for new ones. So it's not recommended to use with branches/commits that are being referenced by others or exists on the remote repository.
+
+Because when we are rebasing or squashing commits that anyone depends on and were not yet pushed to the remote repository, it will have no side effects. But, if the commits exist already on a remote repository and we try push our changes, git will identify that our commits diverge from the commits that exists on the repository and will fail requesting us to get those existing commits from the repository, but that would end of duplicating or maintaining the commits we tried change.
```

Comment:
> But, if the commits exist already on a remote repository and we try push our changes, git will identify that our commits diverge from the commits that exists on the repository and will fail requesting us to get those existing commits from the repository, but that would end of duplicating or maintaining the commits we tried change.
There are ways to force the push to the repository, but that would be too dangerous to be done on a branch used by other people, generating conflicts to everyone else.

We can make this section simpler (with smaller, independent and more direct sentences), what do you think about this?

> When we do rebase/squash and rewrite commits that exists in the remote repository, the local commit tree diverges from the remote tree.
 A _forced push_ (`git push --force`) is needed to push local changes through remote branch, replacing completely some part of the commit history and commit tree.
This _may_ cause losses when rebase/squash is done improperly, and should always be made carefully.
```

In my other suggestion, the part that states the risk of doing it in shared repositories is in the beginning of section, so we do not need to repeat it here.

Reply:
Nice, and we would end of even removing the suggestion of the _forced push_ using only the first paragraph.